builder/builder.go

```diff
@@ -162,18 +162,20 @@ func (b *Builder) RunTask(ctx context.Context, task *graph.Task) error {
 	var deps []*image.Dependencies
 	for _, step := range task.Steps {
 		log.Printf("Step ID: %v marked as %v (elapsed time in seconds: %f)\n", step.ID, step.StepStatus, step.EndTime.Sub(step.StartTime).Seconds())
-		// currently we have a limitation where we cannot fetch the digest of the dependencies
-		// because we build the image using buildx, but query the digest using docker.
-		// https://github.com/Azure/acr-builder/issues/522
-		if step.UseBuildCacheForBuildStep() && runtime.GOOS == util.LinuxOS {
-			continue
-		}
+
 		if len(step.ImageDependencies) > 0 {
 			log.Printf("Populating digests for step ID: %s...\n", step.ID)
 			timeout := time.Duration(digestsTimeoutInSec) * time.Second
 			digestCtx, cancel := context.WithTimeout(ctx, timeout)
 			defer cancel()
-			if err := b.populateDigests(digestCtx, step.ImageDependencies); err != nil {
+
+			usingBuildkit := false
+			if (step.UseBuildCacheForBuildStep() && runtime.GOOS == util.LinuxOS) || step.UsesBuildkit {
+				log.Printf("Image was built using buildkit, fetching Digest from remote...")
+				usingBuildkit = true
+			}
+
+			if err := b.getPopulateDigests(digestCtx, step.ImageDependencies, usingBuildkit, task.RegistryLoginCredentials); err != nil {
```

Comment:
If the base image is from other foreign registry, does `RegistryLoginCredentials` include it?

Reply:
For example, if the customer uses managed identity to login to the foreign registry, can the code handle the case?

Reply:
Yes, they are resolved credentials (opaque, KV, and MSI)